spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/DefaultPubSubAcknowledger.java

```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.pubsub.core;
+
+import java.util.Collection;
+
+import com.google.cloud.pubsub.v1.stub.SubscriberStub;
+import com.google.pubsub.v1.AcknowledgeRequest;
+import com.google.pubsub.v1.ModifyAckDeadlineRequest;
+
+/**
+ * Uses the Google Cloud Pub/Sub client library to (negatively) acknowledge messages based on their
+ * ack IDs and subscription name.
+ *
+ * @author João André Martins
+ */
+public class DefaultPubSubAcknowledger implements PubSubAcknowledger {
```

Comment:
I wouldn't expose this class as public API. This is just our internal contract between `PubSubAcknowledger` and `SubscriberFactory`.

Reply:
I might be missing something, but why do you say this is a public API?
Do you mean to say this should not be public? But then, we can't instantiate it in `SubscriberFactory` and the users can't use it...

`PubSubAcknowledger` should be the public API.

Let me know what I'm missing.

Reply:
I think I lost you completely.
This class must be `public`, like I said, users can't use it if it's not (we instantiate it in `SubscriberFactory` and there is a getter for that class in `PubSubTemplate`, for example...)

I'm OK moving the class to the `support` package, but not sure about `core` not using anything from `support`.
Obviously `PubSubTemplate` needs to use `SubscriberFactory`, `PublisherFactory` and `PubSubMessageConverter`. Likewise, it also needs to use `PubSubAcknowledger`...

I'm very confused by the suggestion, please clarify!